src/tests/Interop/COM/Activator/ComActivationContextShim.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+
+namespace Activator
+{
+    internal sealed class ComActivationContextShim
```

Comment:
Nit: You can create a local `ComActivationContext` definition, get the pointer to the method and invoke it via the pointer. The types do not have to match that way, and it better matches how the method is called in the real shim.